File: live-chat-client-electron/src/services/user-service.js
import axios from "../boot/axios";
import { stringify } from "qs";

const http = axios.axiosInst;

/**
 * 根据ID查询用户信息
 * @param {String} userId
 */
async function findByUserId(userId) {
  return await http.get(`/user/${userId}`);
}

/**
 * 登陆
 * @param {String} key 微信号/手机号/邮箱
 * @param {String} password 输入的密码
 */
async function signIn(key, password) {
  await http.post(
    `/user/sign-in-user`,
    stringify({
      key,
      password
    })
  );
}

/**
 * 注册
 * @param {*} user
 */
async function registry({ username, phoneNumber, password }) {
  await http.post(`/user/registry`, {
    username,
    phoneNumber,
    password
  });
}

export default {
  findByUserId,
  signIn,
  registry
};

File: live-chat-stand-alone/src/main/java/com/github/mrag/livechat/modules/user/service/UserService.java
package com.github.mrag.livechat.modules.user.service;

import com.github.mrag.livechat.modules.user.dto.LivechatUserRegistration;
import com.github.mrag.livechat.modules.user.entity.LivechatUser;
import com.github.mrag.livechat.modules.user.vo.LivechatUserRelationVO;

import java.util.List;

/**
 * 用户相关服务
 */
public interface UserService {
    /**
     * 注册一个账户，注册成功后返回一个令牌
     *
     * @param user 注册信息
     * @return token
     */
    String signUp(LivechatUserRegistration user);

    /**
     * 登陆，成功登陆返回一个令牌
     *
     * @param phone    手机号
     * @param password 密码
     * @return token
     */
    String signIn(String phone, String password);

    /**
     * 获取用户资料
     *
     * @param userId 主键
     * @return 用户资料
     */
    LivechatUser findByUserId(Long userId);

    /**
     * 获取用户关系列表
     *
     * @param userId 主键
     * @return 关系列表
     */
    List<LivechatUserRelationVO> findFriendsByUserId(Long userId);
}
